keps/sig-storage/1847-autoremove-statefulset-pvcs/README.md

```diff
@@ -0,0 +1,681 @@
+creation need not be retained after the StatefulSet is deleted. User also requires that the scale 
+up/down occurs in a fast manner, and leverages any previously existing auto created PVCs within the 
+life time of the StatefulSet. A provision needs to be provided for the user to auto-delete the PVCs 
+once the StatefulSet is deleted. 
+
+User would set the VolumeReclaimPolicy as 'RemoveOnStatefulSetDelete' which would ensure that 
+the PVCs created automatically during the StatefulSet activation is removed once the StatefulSet 
+is deleted.
+
+#### Story 2
+User is cost conscious but at the same time can sustain slower scale up(after a scale down) speeds. Needs 
+a provision where the PVC created for a pod(which is part of the StatefulSet) is removed when the Pod 
+is deleted as part of a scale down. Since the subsequent scale up needs to create fresh PVCs, it will 
+be slower than scale ups relying on existing PVCs(from earlier scale ups). 
+
+User would set the VolumeReclaimPolicy as 'RemoveOnPodDeletion' ensuring PVCs are deleted when corresponding
+Pods are deleted. New Pods created during sclae up followed by a scaledown will wait for freshly created PVCs.
+
+### Notes/Constraints/Caveats (optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable.  This may include API specs (though not always
+required) or even code snippets.  If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Volume reclaim policy for the StatefulSet created PVCs
+A new field named `VolumeReclaimPolicy` of the type `StatefulSetVolumeReclaimPolicy` 
+will be added to the StatefulSet. This field represents the user indication on whether the 
+associated PVCs can be deleted or not. If this field is not set, the default behaviour is 
+to leave the PVCs as is during the StatefulSet deletion.
+
+If `VolumeReclaimPolicy` is set to `RemoveOnPodDeletion` Pod is set as the owner of the PVC. 
+When a Pod is deleted, the PVC corresponding to the Pod also gets deleted. During scale up, a 
+check is made to see if the ownership is already set and if the Pod UID is that of the current 
+Pod in consideration. In case of a mismatch the scale up will exit out of the current reconcile 
```

Comment:
This is a little unclear to me. Is "Pod UID" the pod UID in the owner reference of the PVC? and what is "the current Pod in consideration"---wouldn't the controller be looking at all pods & pvcs at once?

I think it might be more clear to just state the behavior rather than saying what goes wrong with different behavior, something like:

When a Pod is deleted, the PVC owned by the Pod is also deleted. During scale-up, for each Pod of the StatefulSet, the PVCs are gathered. If a PVC has an OwnerRef that does not match this Pod, the PVC must not have yet been deleted and the controller will exit the current reconcile loop and attempt to reconcile in the next iteration. This avoids a race with PVC deletion.